server/datastore/mysql/migrations/tables/20240701113709_VPPDBUpdates.go

```diff
@@ -71,7 +71,8 @@ CREATE TABLE vpp_apps_teams (
 	team_id INT(10) UNSIGNED NOT NULL,
 	global_or_team_id INT(10) NOT NULL DEFAULT 0,
 	FOREIGN KEY (adam_id) REFERENCES vpp_apps (adam_id) ON DELETE CASCADE,
-	FOREIGN KEY (team_id) REFERENCES teams (id) ON DELETE CASCADE
+	FOREIGN KEY (team_id) REFERENCES teams (id) ON DELETE CASCADE,
+	UNIQUE KEY idx_adam_id_team_id (adam_id, team_id)
```

Comment:
Heads-up, you'll want the unique key on `global_or_team_id, adam_id`, so that no-team entries are also unique (otherwise for no-team, team_id is NULL and you can get duplicates, because NULL values are never considered the same so the unique constraint won't fail). Also, I'd put the `global_or_team_id` first as this will also serve as an index when using `WHERE global_or_team_id = ?`, which I'd think is more likely than searching for a specific `adam_id`.

Reply:
trying to do too many things at once!